### PR TITLE
Update section, add expanded content (close #164)

### LINKED ISSUE #164: Update 4.2.1, add content on search engine recon
# Issue

There are references to things that no longer exist.

Search operators and search engines are lumped together, and search operators are not well described. This section would benefit from separation and expansion.

### PR DESCRIPTION
**Pull request by victoriadrake**
- Expand content re: search engines and how they differ, search operators, Google dorking.
- Remove instructions about "clicking the 'Cached' next to each Google Search Result" as this is a UI element and has since moved. Instead favor the search operator `cache:`.
- Link to websites, not zip files.
- Remove link to inactive OWASP Project.
- Remove link to [extinguished PunkSpider](https://twitter.com/LaNMaSteR53/status/1086342749849354240?s=20).
- Update screenshots.
- Minor copy edits.

This PR covers issue #164.

- [x] This PR handles the issue and requires no additional PRs.
- [x] You have validated the need for this change.

### CODE REVIEW

File: document/4_Web_Application_Security_Testing/4.2_Information_Gathering/4.2.1_Conduct_Search_Engine_Discovery_Reconnaissance_for_Information_Leakage_OTG-INFO-001.md [hunk 1]
```diff
@@ -4,61 +4,71 @@
...
+- usernames, passwords, and private keys;
+- third-party, or cloud service configuration files;
+- revealing error message content; and
+- development, test, user acceptance testing (UAT), and staging versions of the website.
 
-### Search Operators
+### Search Engines
 
-Using the advanced “site:” search operator, it is possible to restrict search results to a specific domain [[2]]. Do not limit testing to just one search engine provider as they may generate different results depending on when they crawled content and their own algorithms. Consider using the following search engines:
+Do not limit testing to just one search engine provider. Different search engines may generate different results, depending on when they crawled content, and their own algorithms. Consider using the following (alphabetically-listed) search engines:
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.2_Information_Gathering/4.2.1_Conduct_Search_Engine_Discovery_Reconnaissance_for_Information_Leakage_OTG-INFO-001.md]: This sentence is a bit heavy to read `Different search engines may generate different results, depending on when they crawled content, and their own algorithms`. Since we are working at this, can we maybe reword this a little bit to deliver the intended message?

File: document/4_Web_Application_Security_Testing/4.2_Information_Gathering/4.2.1_Conduct_Search_Engine_Discovery_Reconnaissance_for_Information_Leakage_OTG-INFO-001.md [hunk 2]
```diff
@@ -4,61 +4,71 @@
...
+- [binsearch.info](https://binsearch.info/), a search engine for binary Usenet newsgroups.
+- [DuckDuckGo](https://duckduckgo.com/), a privacy-focused search engine that compiles results from many different [sources](https://help.duckduckgo.com/results/sources/). Supports [search syntax](https://help.duckduckgo.com/duckduckgo-help-pages/results/syntax/).
+- [Google](https://www.google.com/), which offers the world's most popular search engine [[2]], and uses a ranking system to attempt to return the most relevant results. Supports [search operators](https://support.google.com/websearch/answer/2466433).
+- [Startpage](https://www.startpage.com/), a search engine that uses Google's results without collecting personal information through trackers and logs. Supports [search operators](https://support.startpage.com/index.php?/Knowledgebase/Article/View/989/0/advanced-search-which-search-operators-are-supported-by-startpagecom).
+- [Shodan](https://www.shodan.io/), a paid service for searching Internet-connected devices and services.
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.2_Information_Gathering/4.2.1_Conduct_Search_Engine_Discovery_Reconnaissance_for_Information_Leakage_OTG-INFO-001.md]: Mentioning that it is a paid service removes the possibility of it being free as well, which it is. Mention that it is free for a small amount of results returned

File: document/4_Web_Application_Security_Testing/4.2_Information_Gathering/4.2.1_Conduct_Search_Engine_Discovery_Reconnaissance_for_Information_Leakage_OTG-INFO-001.md [hunk 3]
```diff
@@ -79,11 +91,5 @@ Periodically review the sensitivity of existing design and configuration informa
 
 ## References
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.2_Information_Gathering/4.2.1_Conduct_Search_Engine_Discovery_Reconnaissance_for_Information_Leakage_OTG-INFO-001.md]: Haha please remove this section. The links in `[X]` don't appear in MD.

Comment (kingthorin) [reply document/4_Web_Application_Security_Testing/4.2_Information_Gathering/4.2.1_Conduct_Search_Engine_Discovery_Reconnaissance_for_Information_Leakage_OTG-INFO-001.md]: In fact if the links can be in-lined in a reasonable manner that's preferred vs. the old footnote style.

Comment (victoriadrake) [reply document/4_Web_Application_Security_Testing/4.2_Information_Gathering/4.2.1_Conduct_Search_Engine_Discovery_Reconnaissance_for_Information_Leakage_OTG-INFO-001.md]: Well I got the syntax wrong, but I thought we were going for [markdown reference links](https://www.markdownguide.org/basic-syntax/#reference-style-links). I prefer the inline style myself anyway, so I'll change this.

Comment (kingthorin) [reply document/4_Web_Application_Security_Testing/4.2_Information_Gathering/4.2.1_Conduct_Search_Engine_Discovery_Reconnaissance_for_Information_Leakage_OTG-INFO-001.md]: I think you actually had the syntax correct, it's just that when they're done that way they don't actually show at the bottom of the page/post which makes the heading irrelevant/removable.

image

**Comment by victoriadrake** [comment]
Added some ELI5 to support the Summary, other changes per review.

File: document/4_Web_Application_Security_Testing/4.2_Information_Gathering/4.2.1_Conduct_Search_Engine_Discovery_Reconnaissance_for_Information_Leakage_OTG-INFO-001.md [hunk 4]
```diff
@@ -2,9 +2,11 @@
 
 ## Summary
 
-There are direct and indirect elements to search engine discovery and reconnaissance. Direct methods relate to searching the indexes and the associated content from caches. Indirect methods relate to gleaning sensitive design and configuration information by searching forums, newsgroups, and tendering websites.
+In order for search engines to work, computer programs (or "robots") regularly fetch data (referred to as "crawling) from billions of pages on the web. These programs find web pages by following links from other pages, or by looking at sitemaps. If a website uses a special file called "robots.txt" to list pages that it does not want search engines to fetch, then the pages listed there will be ignored. This is a basic overview - Google offers a more in-depth explanation of [how a search engine works](https://support.google.com/webmasters/answer/70897?hl=en).
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.2_Information_Gathering/4.2.1_Conduct_Search_Engine_Discovery_Reconnaissance_for_Information_Leakage_OTG-INFO-001.md]: `"crawling` -> `[crawling](https://en.wikipedia.org/wiki/Web_crawler)`? What do you think about linking to what crawling is instead of setting it in quotes? Based on the time I have got Web crawler, if you find anything better, you can certainly replace it with that 😄

File: document/4_Web_Application_Security_Testing/4.2_Information_Gathering/4.2.1_Conduct_Search_Engine_Discovery_Reconnaissance_for_Information_Leakage_OTG-INFO-001.md [hunk 5]
```diff
@@ -2,9 +2,11 @@
...
 
-Once a search engine robot has completed crawling, it commences indexing the web page based on tags and associated attributes, such as `<TITLE>`, in order to return the relevant search results [[1]]. If the robots.txt file is not updated during the lifetime of the web site, and inline HTML meta tags that instruct robots not to index content have not been used, then it is possible for indexes to contain web content not intended to be included by the owners. Website owners may use the previously mentioned robots.txt, HTML meta tags, authentication, and tools provided by search engines to remove such content.
+Testers can use search engines to perform reconnaissance on websites and web applications. There are direct and indirect elements to search engine discovery and reconnaissance: direct methods relate to searching the indexes and the associated content from caches, while indirect methods relate to gleaning sensitive design and configuration information by searching forums, newsgroups, and tendering websites.
```
Comment (ThunderSon) [inline document/4_Web_Application_Security_Testing/4.2_Information_Gathering/4.2.1_Conduct_Search_Engine_Discovery_Reconnaissance_for_Information_Leakage_OTG-INFO-001.md]: Might we be able to replace `geaning`? Pretty sure non native English readers will have trouble with it.
